exp/internal/controllers/machinepool_controller_phases_test.go

```diff
@@ -1200,9 +1201,11 @@ func TestReconcileMachinePoolScaleToFromZero(t *testing.T) {
 		g.Expect(machinepool.Status.GetTypedPhase()).To(Equal(expv1.MachinePoolPhaseRunning))
 
 		delNode := &corev1.Node{}
-		err = env.Get(ctx, client.ObjectKeyFromObject(node), delNode)
-		g.Expect(err).ToNot(BeNil())
-		g.Expect(apierrors.IsNotFound(err)).To(BeTrue())
+
+		g.Eventually(func() bool {
+			err = env.Get(ctx, client.ObjectKeyFromObject(node), delNode)
```

Comment:
what about using `env.GetAPIReader().Get(...)` instead? This would bypass the cache and should be faster than `env.Get(...)`

Reply:
If it's just about the cache this might be easier (vs. it being an async operation, but looks like it's sync)

Reply:
I think that's probably right. I'll update to use the APIReader here.

Reply:
I verified it. It's part of reconcile => reconcileNodeRefs => deleteRetiredNodes l.170.

So it's synchronously called and APIReader should fix it